Parsers/ASimNetworkSession/ProductParsers/ASimNetworkSessionPaloAltoPanOS.yaml

```diff
@@ -0,0 +1,101 @@
+Parser:
+  Title: Network Session ASIM parser for Palo Alto Networks
+  Version: '0.2'
+  LastUpdated: Dec 20, 2021
+Product:
+  Name: PaloAlto
+Normalization:
+  Schema: NetworkSessions
+  Version: '0.2.1'
+References:
+  - Title: ASIM Network Session Schema
+    Link: https://aka.ms/ASimNetworkSessionDoc
+  - Title: ASIM
+    Link: https://aka.ms/AboutASIM
+  - Title: traffic log fields
+    Link: https://docs.paloaltonetworks.com/pan-os/10-1/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/traffic-log-fields.html
+  - Title: Palo Alto Common Event Format Integration Guide [pdf]
+    Link: https://docs.paloaltonetworks.com/content/dam/techdocs/en_US/pdf/cef/pan-os-90-cef-configuration-guide.pdf
+Description: |
+  This ASIM parser supports normalizing Palo Alto Networks logs produced by the Microsoft Sentinel Palo Alto Networks connector to the ASIM Network Session normalized schema.
+ParserName: ASimNetworkSessionPaloAltoCEF
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let Actions=datatable(DeviceAction:string,DvcAction:string)
+    [ "reset client","Reset Source"
+    , "reset server","Reset Destination"
+    , "reset both", "Reset"
+    , "allow","Allow"
+    , "deny","Deny"
+    , "drop", "Drop"
+    , "drop ICMP", "Drop ICMP"];
+  let NWParser=(disabled:bool=false){
+  CommonSecurityLog | where not(disabled)
+  | where DeviceVendor=="Palo Alto Networks" and (Activity=="TRAFFIC")
+  | parse AdditionalExtensions with "PanOSDstPackets="DstPackets:long";PanOSsrc_packets="SrcPackets:long";start="EventStartTime:datetime";reason="*
+  | project-rename 
+        EventProduct=DeviceProduct // Not Documented
+      , EventProductVersion=DeviceVersion // Not Documented
+      , Dvc=DeviceName   
+      , NetworkApplicationProtocol=ApplicationProtocol
+      , SrcZone=DeviceCustomString4 
+      , DstZone=DeviceCustomString5
+      , NetworkRuleName=DeviceCustomString1
+      , NetworkProtocol=toupper(Protocol)
+      , SrcBytes=SentBytes
+      , DstBytes=ReceivedBytes 
+      , SrcUsername=SourceUserName 
+      , DstUsername=DestinationUserName 
+      , EventOriginalSeverity=LogSeverity // not documented
+      , NetworkPackets=DeviceCustomNumber2 
+      , SrcNatIpAddr=SourceTranslatedAddress
+      , DstNatIpAddr=DestinationTranslatedAddress
+      , PaloAltoFlags=FlexString1 // Flags
+  | extend
+    EventVendor="Palo Alto Networks"
+      , NetworkBytes=tolong(FlexNumber1)
+      , SrcUsernameType=case(isempty(SrcUsername), "", SrcUsername contains "@", "UPN", "Simple")
+      , DstUsernameType=case(isempty(DstUsername), "", DstUsername contains "@", "UPN", "Simple")
+      , EventType="NetworkSession"
+      , EventCount=toint(1)
+      , EventResult=case(DeviceAction=="allow","Success","Failure")
+      , NetworkSessionId=tostring(DeviceCustomNumber1)
+      , NetworkDuration=1000*DeviceCustomNumber3
```

Comment:
You might want toint here (I don't remember the original type of CustomNumber)

Reply:
DeviceCustomNumberXs are int.

Reply:
Thanks